### PR TITLE
Zolk 20 hadoop catalog and dependency updates

### PR DESCRIPTION
**Pull request by wobu**
#### Description

Reworked Plugin packaging to include default hadoop resources. This allows to use HDFS as a target with HadoopCatalog.
Also Updated Iceberg to 1.0.0, Spark to 3.3 runtime and dependency of Kafka Connect API to 3.2.3

Tested it with a Strimzi Kafka Connect installation
Unit tests run successfully

##### PR Checklist
- [ ] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,22 +6,22 @@
 
     <groupId>com.getindata</groupId>
     <artifactId>kafka-connect-iceberg-sink</artifactId>
-    <version>0.1.3-SNAPSHOT</version>
+    <version>0.1.4-SNAPSHOT</version>
```
Comment (gliter) [inline pom.xml:9]: Version is updated autmatically during preapre release action:
https://github.com/getindata/kafka-connect-iceberg-sink/blob/36523e9b7c72529b921dbccb7fc8e4504b9bca22/.github/workflows/prepare-release.yml#L28

File: src/test/java/com/getindata/kafka/connect/iceberg/sink/IcebergSinkSystemTest.java [hunk 2]
```diff
@@ -73,8 +66,9 @@ class IcebergSinkSystemTest {
             .withKafkaBootstrap(kafkaContainer.getInternalBootstrap())
             .withPlugin(getJarPath());
 
-    private static final HttpClient httpClient = HttpClient.newHttpClient();;
-    private static SparkTestHelper sparkTestHelper ;
+    private static final HttpClient httpClient = HttpClient.newHttpClient();
+    ;
```
Comment (gliter) [inline src/test/java/com/getindata/kafka/connect/iceberg/sink/IcebergSinkSystemTest.java:70]: Could I ask you to remove additional semicolon?

**Comment by gliter** [comment]
Hi, looks good. Sorry for late response I was on vacations.

File: src/test/java/com/getindata/kafka/connect/iceberg/sink/IcebergSinkSystemTest.java (hunk 2 above)
Comment (wobu) [reply src/test/java/com/getindata/kafka/connect/iceberg/sink/IcebergSinkSystemTest.java]: changed in the new commit

File: pom.xml (hunk 1 above)
Comment (wobu) [reply pom.xml]: changed in the new commit

**Comment by gliter** [comment]
@wobu build fails due to output path change.
```
 Error:  Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.3:run (unpack-assembly) on project kafka-connect-iceberg-sink: An Ant BuildException has occured: Error while expanding /home/runner/work/kafka-connect-iceberg-sink/kafka-connect-iceberg-sink/target/kafka-connect-iceberg-sink-0.1.4-SNAPSHOT-plugin.zip
Error:  java.io.FileNotFoundException: /home/runner/work/kafka-connect-iceberg-sink/kafka-connect-iceberg-sink/target/kafka-connect-iceberg-sink-0.1.4-SNAPSHOT-plugin.zip (No such file or directory)
```
Please also check if paths in build actions are not needed to be change as well.

**Comment by gliter** [comment]
There are still some errors during build. Does it pass locally for you?

**Comment by wobu** [comment]
not anymore, i'll investigate it later and fix it

**Comment by gliter** [comment]
@wobu looks like some property was missing. Fix was just merged. 
https://github.com/getindata/kafka-connect-iceberg-sink/pull/10#issuecomment-1315227751

**Comment by wobu** [comment]
merged upstream develop branch, but tests still failing. will investigate later

**Review by gliter — APPROVED** [review]

**Pull request merged**